### PR TITLE
DCS-1263 implemented cookie to pass imprisonmentStatus and movementReason data

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/routes/checkAnswersController.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
 import { RequestHandler } from 'express'
 import { Gender } from 'welcome'
 import ExpectedArrivalsService from '../services/expectedArrivalsService'
+import ImprisonmentStatusesService from '../services/imprisonmentStatusesService'
```
Comment (andrewrlee) [inline server/routes/checkAnswersController.ts:4]: could be importing the type of these services

File: server/routes/imprisonmentStatusesController.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { RequestHandler } from 'express'
 import ImprisonmentStatusesService from '../services/imprisonmentStatusesService'
 import ExpectedArrivalsService from '../services/expectedArrivalsService'
```
Comment (andrewrlee) [inline server/routes/imprisonmentStatusesController.ts:3]: import types?

File: integration_tests/integration/checkAnswers.spec.ts [hunk 3]
```diff
@@ -28,13 +29,18 @@ context('Check Answers', () => {
   it('Should contain a full set of correctly formatted move data', () => {
     cy.task('stubExpectedArrival', expectedArrival)
     cy.signIn()
+    cy.setCookie(
```
Comment (andrewrlee) [inline integration_tests/integration/checkAnswers.spec.ts:32]: you shouldn't need to set the cookie for these integration tests - just navigating through the application should set and read the application appropriately.

File: server/routes/confirmAddedToRollController.test.ts [hunk 4]
```diff
@@ -76,6 +76,21 @@ describe('GET /confirmation', () => {
...
+      .get('/prisoners/12345-67890/confirmation')
+      .expect(res => {
+        expect(res.header['set-cookie'][0]).not.toContain('code')
+      })
+      .expect(res => {
+        expect(res.header['set-cookie'][0]).not.toContain('imprisonmentStatus')
+      })
+      .expect(res => {
+        expect(res.header['set-cookie'][0]).not.toContain('movementReasonCode')
+      })
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/confirmAddedToRollController.test.ts:91]: Sorted for next commit

File: server/routes/checkAnswersController.ts (hunk 1 above)
Comment (louisenorris) [reply server/routes/checkAnswersController.ts:4]: Sorted for next commit

File: server/routes/imprisonmentStatusesController.ts (hunk 2 above)
Comment (louisenorris) [reply server/routes/imprisonmentStatusesController.ts:3]: Sorted for next commit

File: integration_tests/integration/checkAnswers.spec.ts (hunk 3 above)
Comment (louisenorris) [reply integration_tests/integration/checkAnswers.spec.ts:32]: Sorted for next commit

File: server/routes/checkAnswersController.test.ts [hunk 5]
```diff
@@ -29,89 +42,118 @@ beforeEach(() => {
...
+      return request(app)
+        .get('/prisoners/12345-67890/check-answers')
+        .set(
+          'cookie',
+          `status-and-reason=s%3Aj%3A%7B%22code%22%3A%22determinate-sentence%22%2C%22imprisonmentStatus%22%3A%22SENT%22%2C%22movementReasonCode%22%3A%2226%22%7D.QEx%2B2EcyCfMkSknBJwkaVswIBLsUTbGFLkXur2qN%2Fro; Max-Age=7200; Domain=localhost; Path=/; Expires=${hourFromNow}; HttpOnly; SameSite=Lax`
+        )
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(res => {
+          expect(expectedArrivalsService.getMove).toHaveBeenCalledTimes(1)
+          expect(expectedArrivalsService.getMove).toHaveBeenCalledWith('12345-67890')
```
Comment (andrewrlee) [inline server/routes/checkAnswersController.test.ts:73]: meh.. future reference - don't need to do both

```
          expect(expectedArrivalsService.getMove).toHaveBeenCalledTimes(1)
          expect(expectedArrivalsService.getMove).toHaveBeenCalledWith('12345-67890')
```

just the calledWith version is enough, its probably over specifying the behaviour saying how many times its being called. calledWith will fail if not called at all

**Review by andrewrlee — APPROVED** [review]

File: server/routes/checkAnswersController.test.ts (hunk 5 above)
Comment (louisenorris) [reply server/routes/checkAnswersController.test.ts:73]: Cool, I'll remove the toHaveBeenCalledTimes as a tidy up in the next PR

**Pull request merged**